moveit_core/collision_detection/CMakeLists.txt

```diff
@@ -27,48 +28,49 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
 
 # unit tests
 if(BUILD_TESTING)
-	find_package(resource_retriever REQUIRED)
-	if(WIN32)
-		# set(append_library_dirs "$<TARGET_FILE_DIR:${PROJECT_NAME}>;$<TARGET_FILE_DIR:${PROJECT_NAME}_TestPlugins1>")
-	else()
-		set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../utils")
-	endif()
+  find_package(resource_retriever REQUIRED)
+  if(WIN32)
+    # TODO add window paths
+  else()
+   set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../utils")
```

Comment:
Hardcoding relative paths is a bad idea. 

I would recommend that for every library in this package that is used else ware you set a `<lib_name>_lib_dir` environment variable that can then be used by the tests

A good example is the `rcl_lib_dir` I have taken the time to show how it is used below:
```
/home/mike/ws_ros2/src/ros2/rcl/rcl/CMakeLists.txt:
   78  
   79: # rcl_lib_dir is passed as APPEND_LIBRARY_DIRS for each ament_add_gtest call so
   80  # the librcl that they link against is on the library path.
   81  # This is especially important on Windows.
   82  # This is overwritten each loop, but which one it points to doesn't really matter.
   83: set(rcl_lib_dir "$<TARGET_FILE_DIR:${PROJECT_NAME}>")
   84  

/home/mike/ws_ros2/src/ros2/rcl/rcl/test/CMakeLists.txt:
   16  
   17: set(extra_lib_dirs "${rcl_lib_dir}")
   18  
   
/home/mike/ws_ros2/src/ros2/rcl/rcl/test/CMakeLists.txt:
   34    rcl_add_custom_gtest(test_client${target_suffix}
   35      SRCS rcl/test_client.cpp
   36:     INCLUDE_DIRS ${osrf_testing_tools_cpp_INCLUDE_DIRS}
   37      ENV ${rmw_implementation_env_var}
   38      APPEND_LIBRARY_DIRS ${extra_lib_dirs}
```

The example uses rcl_add_custom_gtest but it works just as well with ament_add_gtest

Reply:
If you don't mind I prefer to merge this. And then I will fix them all together. Because we have other merged that use this style. I can open an issue to avoid forgetting it.

Reply:
Please open an issue

Reply:
Opened one https://github.com/ros-planning/moveit2/issues/91